actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchNotificationActor.java

```diff
@@ -0,0 +1,421 @@
+      Set<String> keys = participants.keySet();
+      for (String user : keys) {
+        if (participants.get(user)) {
+          usersIds.add(user);
+        }
+      }
+    }
+    return usersIds;
+  }
+
+  /*
+   * This method takes List<String>userIds and returns List<Map<User,Objects>> of
+   * participants
+   *
+   * @param List<String> userIds list of userIds
+   *
+   * @return List<Map<String, Object>> map of user related data.
+   */
+  private List<Map<String, Object>> getUsersFromDB(List<String> userIds) {
+    Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+    List<String> userIdList = new ArrayList<>(userIds);
+    List<String> fields = new ArrayList<>();
+    fields.add(JsonKey.FIRST_NAME);
+    fields.add(JsonKey.EMAIL);
+    Response response =
+        cassandraOperation.getRecordsByIdsWithSpecifiedColumns(
+            usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), fields, userIdList);
+    if (response == null) {
+      ProjectLogger.log("No data from cassandra , check connection  ", LoggerEnum.ERROR.name());
+    }
+    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    return userList;
+  }
+
+  /*
+   * This method takes template name and return template.
+   *
+   * @param templateName String
+   *
+   * @return email template String .
+   */
+  private String getEmailTemplateFile(String templateName) {
+    EmailTemplateDao emailTemplateDao = EmailTemplateDaoImpl.getInstance();
+    String template = emailTemplateDao.getTemplate(templateName);
+    if (StringUtils.isBlank(template)) {
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.invalidParameterValue,
+          MessageFormat.format(
+              ResponseCode.invalidParameterValue.getErrorMessage(),
+              templateName,
+              JsonKey.EMAIL_TEMPLATE_TYPE));
+    }
+    return template;
+  }
+
+  public static VelocityContext getContext(Map<String, Object> map) {
+    VelocityContext context = new VelocityContext();
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.COURSE_NAME))) {
+      context.put(JsonKey.COURSE_NAME, getValue(map, JsonKey.COURSE_NAME));
+    }
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.START_DATE))) {
+      context.put(JsonKey.BATCH_START_DATE, getValue(map, JsonKey.START_DATE));
+    }
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.END_DATE))) {
+      context.put(JsonKey.BATCH_END_DATE, getValue(map, JsonKey.END_DATE));
+    }
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.NAME))) {
+      context.put(JsonKey.BATCH_NAME, getValue(map, JsonKey.NAME));
+    }
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.FIRST_NAME))) {
+      context.put(JsonKey.NAME, getValue(map, JsonKey.FIRST_NAME));
+    } else {
+      context.put(JsonKey.NAME, "");
+    }
+    return context;
+  }
+
+  private static Object getValue(Map<String, Object> map, String key) {
```

Comment:
No need to define getValue method as map.remove already provides equivalent functionality.

Object value = map.remove(key)